internal/engine/experiment/dnsping/dnsping.go

```diff
@@ -132,54 +132,54 @@ func (m *Measurer) Run(
 	numResults := int(m.config.repetitions()) * len(domains) * 2
 	for len(tk.Pings) < numResults {
 		meas := <-out
+		queries := meas.Queries
 		// TODO(bassosimone): when we merge the improvements at
 		// https://github.com/bassosimone/websteps-illustrated it
 		// will become unnecessary to split with query type
 		// as we're doing below.
-		queries := measurex.NewArchivalDNSLookupEventList(meas.LookupHost)
-		tk.Pings = append(tk.Pings, m.onlyQueryWithType(queries, "A")...)
-		tk.Pings = append(tk.Pings, m.onlyQueryWithType(queries, "AAAA")...)
-	}
-	return nil // return nil so we always submit the measurement
-}
-
-// onlyQueryWithType returns only the queries with the given type.
-func (m *Measurer) onlyQueryWithType(
-	in []*measurex.ArchivalDNSLookupEvent, kind string) (out []*SinglePing) {
-	for _, query := range in {
-		if query.QueryType == kind {
-			out = append(out, &SinglePing{
-				Queries: []*measurex.ArchivalDNSLookupEvent{query},
+		for _, query := range queries {
+			tk.Pings = append(tk.Pings, &SinglePing{
+				Queries: []*model.ArchivalDNSLookupResult{query},
```

Comment:
Would it be possible to fill `tk.Pings` at line `134`? I think we don't need to split the queries anymore, after your changes, so also my TODO comment can now be removed. I think this whole loop at the end is now unnecessary.

Reply:
Introducing the new experiment pattern which helps with this: https://github.com/ooni/probe-cli/pull/831/commits/2ee2eb906e91604b39c1c9985953af8d56b05725